src/workerd/api/actor-state.c++

```diff
@@ -939,6 +939,41 @@ kj::Maybe<kj::Date> DurableObjectState::getWebSocketAutoResponseTimestamp(jsg::R
   return ws->getAutoResponseTimestamp();
 }
 
+void DurableObjectState::setHibernatableWebSocketEventTimeout(jsg::Optional<uint32_t> timeoutMs) {
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+
+  // Setting a timeout = 0ms or an empty value will unset any currently set event timeout.
+  // If there's no hibernation manager instantiated, we can skip the event timeout unsetting.
+  if (timeoutMs == kj::none || KJ_REQUIRE_NONNULL(timeoutMs) == 0) {
+    KJ_IF_SOME(hibernationManager, a.getHibernationManager()) {
+      hibernationManager.setEventTimeout(kj::none);
+    }
+    return;
+  }
+
+  auto t = timeoutMs.orDefault((uint32_t)0);
+
+  // We want to limit the duration of an event to a maximum of 7 days (604800 * 1000 millis).
+  JSG_REQUIRE(t <= 604800 * 1000, Error, "Event timeout should not exceed 604800000 ms.");
+
+  if (a.getHibernationManager() == kj::none) {
+    // If there's no hibernation manager created yet, we should create one and
+    // set its auto response.
```

Comment:
I guess this comment was copied from the earlier code dealing with auto responses...  If we don't actually need to deal with autoresponses here, could update or omit the comment?